### ISSUE TITLE
test occasionally fails

### ISSUE THREAD

**Issue by lloydbenson** [hapijs/hapi#2414]
When running npm test on hapi locally, I see the following test fail occasionally:

  344) Request _execute() handles aborted requests:
          
                Timed out (2000ms) - Request _execute() handles aborted requests

This doesn't happen all the time.  I chatted with @cjihrig as well and he said he noticed this on 0.10 latest (I noticed it when I was using 0.12).

**Comment by arb** [comment]
I think we had this problem before when something in Node changed how timers handle non-whole numbers.

**Comment by hueniverse** [comment]
If you can reproduce it , see if changing the timing of the test helps. Otherwise, not much I can do.

**Comment by lloydbenson** [comment]
If you don't feel this is useful obviously feel free to close this up.  I was able to dig up a little more data.  If I add a console.log to the tests line where it counts the tests, it occassionally skips from 1 to -1 rather than going to 0.  This causes it to continue counting down until the test times out so changing the timing didn't help (I did try that initially but had the same results).  It is helpful I am able to reproduce this more consistently with 0.12.  It took me less than 10 times to reproduce.

Here is where I added the console.log(tests):
https://github.com/hapijs/hapi/blob/master/test/request.js#L225

bad count:
2
2
2
2
2
2
1
-1
-2
-3

normal count:

2
2
2
2
2
1
0

I have added a PR if you think it a bug specific with tests and not a deeper issue.
